ql/src/java/org/apache/hadoop/hive/ql/io/orc/OrcFileStripeMergeRecordReader.java

```diff
@@ -80,7 +80,7 @@ public boolean next(OrcFileKeyWrapper key, OrcFileValueWrapper value) throws IOE
   }
 
   protected boolean nextStripe(OrcFileKeyWrapper keyWrapper, OrcFileValueWrapper valueWrapper)
-      throws IOException {
+        throws IOException {
     // missing stripe stats (old format). If numRows is 0 then its an empty file and no statistics
     // is present. We have to differentiate no stats (empty file) vs missing stats (old format).
     if ((stripeStatistics == null || stripeStatistics.isEmpty()) && reader.getNumberOfRows() > 0) {
```

Comment:
would it make sense to have the ```start > 0``` check here instead?

Reply:
This needs stripe level data and cannot be decided earlier. The information we have is at split level, which can be decided without having to read the file and is hence better to do it constructor. Updating constructor so that reader object is not created if we are skipping file and also changing close to actually close the reader. I think that should not be harmful.